libs/gltfio/src/StbProvider.cpp

```diff
@@ -219,6 +219,18 @@ void StbProvider::cancelDecoding() {
     // in-flight jobs. We should consider throttling the number of simultaneous decoder jobs, which
     // would allow for actual cancellation.
     waitForCompletion();
+
+    // For cancelled jobs, we need to set the TextureInfo to the popped state and free the decoded
+    // data.
+    for (auto& info : mTextures) {
+        if (info->state != TextureState::DECODING) {
+            continue;
+        }
+        if (intptr_t data = info->decodedTexelsBaseMipmap.load()) {
+            delete [] (uint8_t*) data;
+        }
```

Comment:
Is there a risk that another thread deletes this `data` between the `load()` and the `delete[]`? Asking because an atomic is involved. Either way I think a comment about that would be useful here.

Reply:
added comment 
https://github.com/google/filament/pull/6620/files#diff-22acc1b406850d2147dd6e5930f60ea0eb3e38cc31f1493ef5b63d9a5bfb02c5R229